### PR TITLE
Intermediate merge: Spc 836 refactor websocket client api

### PR DESCRIPTION
**Pull request by mvdhart**

### CODE REVIEW

**Review by Li-Yeun — APPROVED** [review]

File: CameraProcessor/processor/websocket/boxes_message.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            message (dict): Python dict representation of an incoming JSON message.
+
+        Returns:
+            (BoxesMessage): BoxesCommand constructed from the dict.
+        """
+        if "frameId" not in message.keys():
+            raise KeyError("frameId missing")
+
+        if "boxes" not in message.keys():
+            raise KeyError("objectId missing")
```
Comment (saberzero1) [inline CameraProcessor/processor/websocket/boxes_message.py:41]: Single qoutes svp

File: CameraProcessor/processor/websocket/boxes_message.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        return BoxesMessage(frame_id, boxes)
+
+    def to_message(self):
+        """Converts the BoxesMessage to a dict representation.
+
+        Returns:
+            (dict): Python dict representation of the message.
+        """
+        return {
+            "type": "boundingBoxes",
```
Comment (saberzero1) [inline CameraProcessor/processor/websocket/boxes_message.py:55]: Same here

File: CameraProcessor/processor/websocket/i_message.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+            message (dict): Python dict representation of an incoming JSON message.
+
+        Returns:
+            (IMessage): IMessage constructed from the dict.
+        """
+        raise NotImplementedError("Expected function from_message not implemented.")
+
+    def to_message(self):
+        """Converts a message object to a dict representation."""
+        raise NotImplementedError("Expected function to_message not implemented.")
```
Comment (saberzero1) [inline CameraProcessor/processor/websocket/i_message.py:26]: Single qoutes

File: CameraProcessor/processor/websocket/start_message.py [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+            object_id (int): Identifier of the object to be followed.
+            image (image/None): URL encoded cutout of the object to be followed.
+            frame_id (int/None): The id of the frame in the frame buffer
+            box_id (int/None): the id of the box in the frame defined by the frame buffer
+        """
+        if not isinstance(object_id, int):
+            raise TypeError("Object id should be an integer")
+
+        if frame_id is not None and not isinstance(frame_id, float):
+            raise TypeError("Frame id should be a float")
```
Comment (saberzero1) [inline CameraProcessor/processor/websocket/start_message.py:34]: Single qoutes

File: CameraProcessor/processor/websocket/stop_message.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        Returns:
+            bool: Whether the messages are the same.
+        """
+        return self.__object_id == other.object_id
+
+    def __repr__(self):
+        """Converts the StopMessage to a string.
+
+        Returns:
+            str: String representation of a StopMessage.
```
Comment (saberzero1) [inline CameraProcessor/processor/websocket/stop_message.py:75]: Single qoutes

File: CameraProcessor/processor/websocket/update_message.py [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+class UpdateMessage(IMessage):
+    """Contains UpdateMessage class which holds an updated feature_map for a followed object."""
+    def __init__(self, object_id, feature_map):
+        """Constructor for the UpdateMessage class.
+
+        Args:
+            object_id (int): Identifier of the object that should be followed."
+            feature_map ([float]): Feature map of the object that should be followed.
+        """
+        if not isinstance(object_id, int):
```
Comment (saberzero1) [inline CameraProcessor/processor/websocket/update_message.py:20]: Single qoutes

File: CameraProcessor/processor/websocket/websocket_client.py [hunk 7]
```diff
@@ -220,82 +208,43 @@ def _on_message(self, message):
...
-            # Switch on message type.
-            actions = {
-                'featureMap':
-                    lambda: self.update_feature_map(message_object),
-                'start':
-                    lambda: self.start_tracking(message_object),
-                'stop':
-                    lambda: self.stop_tracking(message_object)
+            class_dict = {
+                "start": StartMessage,
```
Comment (saberzero1) [inline CameraProcessor/processor/websocket/websocket_client.py:212]: Single qoutes

File: CameraProcessor/tests/orchestratorintegrationtests/test_receiving_from_orchestrator.py [hunk 8]
```diff
@@ -46,6 +43,7 @@ def amount(self, request):
 
     @pytest.mark.asyncio
     @pytest.mark.timeout(60)
+    @pytest.mark.skip("Refactored commands")
```
Comment (saberzero1) [inline CameraProcessor/tests/orchestratorintegrationtests/test_receiving_from_orchestrator.py:46]: Single qoutes

File: CameraProcessor/tests/orchestratorintegrationtests/test_send_to_orchestrator.py [hunk 9]
```diff
@@ -22,6 +22,7 @@ async def test_confirm_connection(self):
         assert ws_client.connection is not None
 
     @pytest.mark.asyncio
+    @pytest.mark.skip("Refactored commands")
```
Comment (saberzero1) [inline CameraProcessor/tests/orchestratorintegrationtests/test_send_to_orchestrator.py:25]: Single qoutes

**Review by saberzero1 — CHANGES_REQUESTED** [review]
The marked files contain qoutations marks rather than single qoutes. Yesterday the all python files were refactored for this. Please fix.

(I have not marked every instance of qoutation marks, but every file with incorrect qoutations has been marked)

**Review by saberzero1 — APPROVED** [review]

**Review by GerardvSchie — APPROVED** [review]

**Pull request merged**